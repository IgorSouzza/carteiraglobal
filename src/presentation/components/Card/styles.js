import { tc } from '../../utils'

export const wrapper = tc(`
  rounded
  w-full
  bg-white
  p-5
  shadow-carteira-card
`)

export const title = tc(`
  text-sm
  font-semibold
  text-carteira-global-green
`)

export const childrenContainer = tc(`
  mt-8
`)
