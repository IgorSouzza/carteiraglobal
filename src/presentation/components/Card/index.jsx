import PropTypes from 'prop-types'

import * as S from './styles'

export default function Card ({ title, circleVariant = false, customClass = '', children }) {
  return (
    <div className={`${S.wrapper({ circleVariant })} ${customClass}`}>
      <h2 className={S.title}>{title}</h2>
      <div className={S.childrenContainer}>
        {children}
      </div>
    </div>
  )
}

Card.propTypes = {
  title: PropTypes.string,
  circleVariant: PropTypes.bool,
  customClass: PropTypes.string,
  children: PropTypes.node
}
