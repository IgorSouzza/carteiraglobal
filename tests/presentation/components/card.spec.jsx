import { render, screen } from '@testing-library/react'
import Card from '../../../src/presentation/components/Card'

describe('<Card />', () => {
  it('should render correctly with children and title', () => {
    const { container } = render(
      <Card title="Card Title">
        <p>Card Children Text</p>
      </Card>
    )

    const title = screen.getByRole('heading', { name: /card title/i })
    const children = screen.getByText(/card children text/i)

    expect(title).toBeInTheDocument()
    expect(children).toBeInTheDocument()
    expect(children.tagName).toBe('P')
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should be able to add a custom class', () => {
    render(
      <Card title="Card Title" customClass="my-custom-class">
        <p>Card Children Text</p>
      </Card>
    )

    const wrapper = screen.getByRole('heading', {
      name: /card title/i
    }).closest('div')

    expect(wrapper).toHaveClass('my-custom-class')
  })
})
