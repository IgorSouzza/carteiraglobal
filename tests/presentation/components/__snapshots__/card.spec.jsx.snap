// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> should render correctly with children and title 1`] = `
<div
  class="w-full bg-white shadow-carteira-card rounded p-5 "
>
  <h2
    class="text-sm font-semibold text-carteira-global-green"
  >
    Card Title
  </h2>
  <div
    class="mt-8"
  >
    <p>
      Card Children Text
    </p>
  </div>
</div>
`;
